intercom-java/src/main/java/io/intercom/api/HttpClient.java

```diff
@@ -237,9 +237,6 @@ private HttpURLConnection prepareConnection(HttpURLConnection conn) {
 
     private Map<String, String> createAuthorizationHeaders() {
         switch (Intercom.getAuthKeyType()) {
```

Comment:
If we're removing `API_KEY` we could also remove this whole switch and just run
`headers.put("Authorization", "Basic " + generateAuthString(Intercom.getToken(),""));`

We can then remove `getAuthKeyType()` in `Intercom` class and get rid of `AuthKeyType` enum altogether.
@thewheat wdyt?

Reply:
I had initial thoughts of keeping it here in case we could have future authentication methods but for simplification, the removal of `getAuthKeyType` and `AuthKeyType` makes sense too 👍 I'll get to it 😄

Reply:
@mmartinic updated 👌